example/example.go

```diff
@@ -0,0 +1,61 @@
+// package example contains a self-contained example of a webhook that passes the cert-manager
+// DNS conformance tests
+package example
+
+import (
+	"sync"
+
+	"github.com/jetstack/cert-manager/pkg/acme/webhook"
+	acme "github.com/jetstack/cert-manager/pkg/acme/webhook/apis/acme/v1alpha1"
+	"github.com/miekg/dns"
+	"k8s.io/client-go/rest"
+)
+
+type exampleSolver struct {
+	name       string
+	server     *dns.Server
+	txtRecords map[string]string
+	sync.RWMutex
+}
+
+func (e exampleSolver) Name() string {
```

Comment:
We are using non pointer receivers here, is that on purpose? Perhaps it is, but this is not the norm so maybe we should add a small comment as to why.

I don't have a huge amount of experience using non pointer receivers, but does this not cause issues with the mutex? (each call re-inits a new mutex which is not shared).

Reply:
That's actually a typo, I'm surprised that works